src/mixer.rs

```diff
@@ -0,0 +1,428 @@
+    pub fn channel_name(&self, channel: i32) -> Result<String> {
+        let c = unsafe { alsa::snd_mixer_selem_channel_name(channel) };
+        from_const("snd_mixer_selem_channel_name", c).and_then(|s| Ok(s.to_string()))
+    }
+
+    pub fn get_playback_volume(&self, channel: i32) -> Result<i64> {
+        let mut value: i64 = 0;
+        acheck!(snd_mixer_selem_get_playback_volume(self.1.handle, channel, &mut value)).and_then(|_| Ok(value))
+    }
+
+    /// returns volume in decibels*100. To get correct dB value, devide by 100
+    ///
+    /// # Example
+    /// ```ignore
+    /// let db_value = selem.playback_volume_decibel(SelemChannelId::FrontLeft as i32).unwrap() as f32 / 100.0;
+    /// ```
+    pub fn ask_playback_vol_decibel(&self, channel: i32) -> Result<i64> {
+        let mut decibel_value: i64 = 0;
+        self.get_playback_volume(channel)
+            .and_then(|volume| acheck!(snd_mixer_selem_ask_playback_vol_dB (self.1.handle, volume, &mut decibel_value)))
+            .and_then(|_| Ok(decibel_value))
+    }
+
+    pub fn get_capture_volume(&self, channel: i32) -> Result<i64> {
+        let mut value: i64 = 0;
+        acheck!(snd_mixer_selem_get_capture_volume(self.1.handle, channel, &mut value)).and_then(|_| Ok(value))
+    }
+
+    /// returns volume in decibels*100. To get correct dB value, devide by 100, i.e.
+    ///
+    /// # Example
+    /// ```ignore
+    /// let db_value = selem.capture_volume_decibel(SelemChannelId::FrontLeft as i32).unwrap() as f32 / 100.0;
+    /// ```
+    pub fn ask_capture_vol_decibel(&self, channel: i32) -> Result<i64> {
+        let mut decibel_value: i64 = 0;
+        self.get_capture_volume(channel)
+            .and_then(|volume| acheck!(snd_mixer_selem_ask_capture_vol_dB (self.1.handle, volume, &mut decibel_value)))
+            .and_then(|_| Ok(decibel_value))
+    }
+
+    pub fn set_playback_volume(&self, channel: i32, value: i64) -> Result<i32> {
+        acheck!(snd_mixer_selem_set_playback_volume(self.1.handle, channel, value))
+    }
+
+    pub fn set_capture_volume(&self, channel: i32, value: i64) -> Result<i32> {
+        acheck!(snd_mixer_selem_set_capture_volume(self.1.handle, channel, value))
+    }
+}
+
+
+#[derive(Copy, Clone)]
+pub enum SelemChannelId {
+    Unknown     = alsa::SND_MIXER_SCHN_UNKNOWN as isize,
+    FrontLeft   = alsa::SND_MIXER_SCHN_FRONT_LEFT as isize,
+    FrontRight  = alsa::SND_MIXER_SCHN_FRONT_RIGHT as isize,
+    RearLeft    = alsa::SND_MIXER_SCHN_REAR_LEFT as isize,
+    RearRight   = alsa::SND_MIXER_SCHN_REAR_RIGHT as isize,
+    FrontCenter = alsa::SND_MIXER_SCHN_FRONT_CENTER as isize,
+    Woofer      = alsa::SND_MIXER_SCHN_WOOFER as isize,
+    SideLeft    = alsa::SND_MIXER_SCHN_SIDE_LEFT as isize,
+    SideRight   = alsa::SND_MIXER_SCHN_SIDE_RIGHT as isize,
+    RearCenter  = alsa::SND_MIXER_SCHN_REAR_CENTER as isize,
+    Last        = alsa::SND_MIXER_SCHN_LAST as isize,
+    // Mono        = alsa::SND_MIXER_SCHN_MONO as isize,
+}
```

Comment:
When there are duplicate constants (such as Front_left and mono), we can provide a mono() function on SelemChannelId instead